ArduPlane/mode_takeoff.cpp

```diff
@@ -68,59 +68,78 @@ bool ModeTakeoff::_enter()
 
 void ModeTakeoff::update()
 {
+    // nothing works if we dont have a valid position!
+    if (!plane.current_loc.initialised()) {
+        return;
+    }
+    plane.prev_WP_loc = plane.current_loc;
+    const float dist = target_dist;
+    const float alt = target_alt;  
     if (!takeoff_started) {
-        // see if we will skip takeoff as already flying
+        const float direction = degrees(ahrs.yaw);
+        // if you are flying but had not used Mode Takeoff before
+        // then start_loc not set and use home altitude as start altitude
+        int32_t alt_cm;
+        if (start_loc.alt == 0 && plane.current_loc.get_alt_cm(Location::AltFrame::ABOVE_HOME, alt_cm)) {
+            start_loc.alt = plane.current_loc.alt - alt_cm;
```

Comment:
I don't understand this. Your trying to get the absolute altitude of home? You could do:
`start_loc.set_alt_cm(0, Location::AltFrame::ABOVE_HOME);`

Reply:
yes trying to set an uninitialized start_loc to home alt so the following conditionals work....does the suggested line do the same? don't need the lat/long to be correct...

Reply:
got it set to home directly  and put it a check to assure there was a home